fire/cli/niv/__init__.py

```diff
@@ -223,39 +224,147 @@ def normaliser_placeringskoordinat(λ: float, φ: float) -> Tuple[float, float]:
 
 # -----------------------------------------------------------------------------
 def skriv_ark(
-    projektnavn: str, resultater: Dict[str, pd.DataFrame], suffix: str = "-resultat"
+    projektnavn: str, nye_faneblade: Dict[str, pd.DataFrame], suffix: str = ""
 ) -> bool:
-    """Skriv resultater til excel-fil"""
+    """Skriv resultater til excel-fil
 
-    filnavn = f"{projektnavn}{suffix}.xlsx"
-    if suffix != "":
-        fire.cli.print(f"Skriver: {tuple(resultater)}")
-        fire.cli.print(f"Til filen '{filnavn}'")
+    Basalt en temmeligt simpel operation, men virkemåden er næsten overskygget af
+    kontroller af at filoperationer gik godt. Det er et vilkår ved interaktioner
+    med filsystemer, så for overblikkets skyld kommer her en kort prosabeskrivelse.
 
-    # Giv brugeren en chance for at lukke et åbent regneark
-    while True:
+    1. Flyt projektnavn.xlsx til projektnavn-ex.xlsx
+    2. Læs dict gamle_faneblade fra projektnavn-ex.xlsx
+    3. Fjern elementer fra gamle_faneblade hvis navnet også findes i nye_faneblade
+    4. Skriv nye_faneblade til projektnavn.xlsx
+    5. Skriv gamle_faneblade til projektnavn.xlsx
+
+    Trin 4 kommer før trin 5 for at sikre at de nye faneblade er umiddelbart
+    synlige når man åbner projektnavn.xlsx
+
+    Eller mere direkte sagt:
+
+    ```
+    fil = Path(f"{projektnavn}{suffix}.xlsx")
+    exfil = Path(f"{projektnavn}{suffix}-ex.xlsx")
+    fil.replace(exfil)
+
+    gamle_faneblade = pd.read_excel(exfil, sheet_name=None)
+    for fanebladnavn in set(gamle_faneblade).intersection(nye_faneblade):
+        gamle_faneblade.pop(fanebladnavn)
+
+    with pd.ExcelWriter(fil) as writer:
+        skriv nye_faneblade
+        skriv gamle_faneblade
+    ```
+
+    Hvilket er i omegnen af en faktor 10 mindre end den implementerede
+    version - men *it's a jungle out there*...
+    """
+
+    fil = Path(f"{projektnavn}{suffix}.xlsx")
+    exfil = Path(f"{projektnavn}{suffix}-ex.xlsx")
```

Comment:
Du kan også overveje at bruge en tempfil for at minimere risikoen for utilsigtet brugerinteraktion med de filer der arbejdes på. Se fx https://stackoverflow.com/a/8577225.

Reply:
Jeg har overvejet tempfile - det bliver temmelig besværligt fordi filhåndteringen klares af `pd.ExcelWriter` og den tager et filnavn, ikke et filhåndtag som argument. Eller, det vil sige: Kigger man på [dokumentationen](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.ExcelWriter.html#pandas.ExcelWriter) så kan den tage snart sagt hvad som helst. Men jeg har kæmpet lidt med ExcelWriter de sidste par dage (en af grundene til at det har taget en krig at nå igennem dette PR)  og der er ting der ser ud til ikke at virke godt under Windows.

Så efter at have spildt megen tid endte jeg med at krybe baglæns i sporet og kun bruge ExcelWriter på simplest mulige måde. Jeg har en stærk fornemmelse af at udviklerne primært sidder på posixplatforme, og jeg er sikker på at det hele fungerer fint med OpenOffice. Vi kan prøve igen efter 2-3 nye udgaver af Pandas. Fordelen er selvfølgelig marginal, men det ærgrer mig ikke at kunne lave al relevant filsystemfifleri med atomiske operationer:

- omdøb `x.xlsx` til `x-ex.xlsx`
- gør arbejdet og skriv det til nyoprettet tempfil
- omdøb tempfil til `x.xlsx`

Tre atomiske operatorer, ingen vrøvl ... men det må blive næste år :-)